### PR TITLE
Refactor minimap creation + new CIC map table

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Title, fixes my shitcode
note the maptable needs to be mapped on other ships

## Why It's Good For The Game
Progress towards map drawing + less shitcode

## Changelog
:cl:
add: The CIC minimap table will now display groundside
refactor: Changed how minimaps were cached
/:cl:

### CODE REVIEW

**Comment by BraveMole** [comment]
But when will be able to draw dongs on the map?

**Comment by Swagile** [comment]
> 
> 
> But when will be able to draw dongs on the map?

asking the real questions here

File: code/controllers/subsystem/minimaps.dm [hunk 1]
```diff
@@ -268,3 +270,117 @@ SUBSYSTEM_DEF(minimaps)
...
+
+/datum/action/minimap/action_activate()
+	. = ..()
+	if(!map)
+		return
+	if(minimap_displayed)
+		owner.client.screen -= map
+	else
+		owner.client.screen += map
+	minimap_displayed = !minimap_displayed
```
Comment (BraveMole) [inline code/controllers/subsystem/minimaps.dm:332]: probably could do that with early return

Comment (TiviPlus) [reply code/controllers/subsystem/minimaps.dm:332]: An early return doesnt apply here

**Review by SplinterGP — APPROVED** [review]

**Pull request merged**